### PR TITLE
fix: Pad angle brackets in user uploaded json or text input to avoid html exception

### PR DESCRIPTION
**Pull request by dsamojlenko**
# Summary | Résumé

This PR addresses an issue where a user could either: 
- upload a form file; or 
- enter text into a question or description input

...that includes HTML or text that looks like HTML (ie `<text that has angle brackets on either end>`, but saving would throw an error and fail to save due to invalid input. This usually results in the user needing to contact support to resolve.

This PR addresses the issue on two ends- on JSON upload, and on input/save, by "escaping" any angle brackets found by adding a space after an opening bracket and a space before a closing bracket.

The matching algorithm only matches strings that contain both an opening and closing bracket, and ignores them if there are already spaces. For example:

- `<something>` becomes `< something >`
- `<something > becomes `< something >`
- `< something>` is valid and remains `< something>`
- `<something` is valid and remains `<something`

The text between the angle brackets can contain any number of alphanumeric characters, spaces, punctuation, or special characters.

Fixes: https://github.com/cds-snc/platform-forms-client/issues/2821

### CODE REVIEW

File: components/form-builder/util.ts [hunk 1]
```diff
@@ -259,3 +259,31 @@ export const isVaultDelivery = (deliveryOption: DeliveryOption | undefined) => {
...
+      } else if (typeof obj[key] === "string") {
+        obj[key] = escapeAngleBrackets(obj[key]);
+      }
+    }
+
+    return obj;
+  }
+
+  return input;
+};
```
Comment (dmethvin-gov) [inline components/form-builder/util.ts:289]: A recursive type can avoid the `any`. I think the code below should work and avoid the `in` operator which is flakey for some objects and arrays.

```suggestion
type Cleanable = string | Cleanable[] | { [key: string]: Cleanable } | unknown;

export const cleanInput = (input: Cleanable) => {
    if (typeof input === "string") {
        return escapeAngleBrackets(input);
    }
    if ( Array.isArray(input) ) {
        return input.map((elem) => cleanInput(elem));
    }
    if ( input instanceof Object ) {
        return Object.fromEntries(
            Object.entries(input).map(
                ([key, value]) => [key, cleanInput(value)]
            )
        )
    }
    return input;
};
};
```

Comment (timarney) [reply components/form-builder/util.ts]: Thanks @dmethvin-gov you beat us to cleaning up the for this :)

That's a nice great improvement on the recursion code 🎉 .

**Comment by timarney** [comment]
@dsamojlenko I fixed up the backspace issue.

Let's get CI fixed up and write some tests for this PR.

**Review by timarney — APPROVED** [review]
There's still room to iterate on this for sure but I think this solves the immediate issue.

Specifically some of the inputs live update while other don't leading to some inconsistency as to what's happening with the fields.  

Will approve `as is` as it should cut down on the incoming support request re:failed to save message.

**Pull request merged**